dask/dataframe/io/tests/test_orc.py

```diff
@@ -75,8 +78,104 @@ def test_orc_single(orc_files):
 
 
 def test_orc_multiple(orc_files):
-    d = read_orc(orc_files[0])
-    d2 = read_orc(orc_files)
+    d = dd.read_orc(orc_files[0])
+    d2 = dd.read_orc(orc_files)
     assert_eq(d2[columns], dd.concat([d, d])[columns], check_index=False)
-    d2 = read_orc(os.path.dirname(orc_files[0]) + "/*.orc")
+    d2 = dd.read_orc(os.path.dirname(orc_files[0]) + "/*.orc")
     assert_eq(d2[columns], dd.concat([d, d])[columns], check_index=False)
+
+
+@pytest.mark.skipif(
+    parse_version(pa.__version__) < parse_version("4.0.0"),
+    reason=("PyArrow>=4.0.0 required for ORC write support."),
+)
+@pytest.mark.parametrize("index", [None, "i32"])
+@pytest.mark.parametrize("columns", [None, ["i32", "i64", "f"]])
+def test_orc_roundtrip(tmpdir, index, columns):
+    tmp = str(tmpdir)
+    data = pd.DataFrame(
+        {
+            "i32": np.arange(1000, dtype=np.int32),
+            "i64": np.arange(1000, dtype=np.int64),
+            "f": np.arange(1000, dtype=np.float64),
+            "bhello": np.random.choice(["hello", "yo", "people"], size=1000).astype(
+                "O"
+            ),
+        }
+    )
+    if index:
+        data.set_index(index, inplace=True)
+    df = dd.from_pandas(data, chunksize=500)
+    if columns:
+        data = data[[c for c in columns if c != index]]
+
+    # Write
+    df.to_orc(tmp, write_index=bool(index))
+
+    # Read
+    df2 = dd.read_orc(tmp, index=index, columns=columns)
+    assert_eq(data, df2, check_index=bool(index))
+
+
+@pytest.mark.skipif(
+    parse_version(pa.__version__) < parse_version("4.0.0"),
+    reason=("PyArrow>=4.0.0 required for ORC write support."),
+)
+@pytest.mark.parametrize("split_stripes", [True, False, 2, 4])
+def test_orc_roundtrip_aggregate_files(tmpdir, split_stripes):
+    tmp = str(tmpdir)
+    data = pd.DataFrame(
+        {
+            "a": np.arange(100, dtype=np.float64),
+            "b": np.random.choice(["cat", "dog", "mouse"], size=100),
+        }
+    )
+    df = dd.from_pandas(data, npartitions=8)
+    df.to_orc(tmp, write_index=False)
+    df2 = dd.read_orc(tmp, split_stripes=split_stripes, aggregate_files=True)
+
+    # Check that we have the expected partition count
+    # and that the data is correct
+    if split_stripes:
+        assert df2.npartitions == df.npartitions / int(split_stripes)
+    else:
+        assert df2.npartitions == df.npartitions
+    assert_eq(data, df2, check_index=False)
+
+
+def test_orc_roundtrip_aggregate_files_offset(tmpdir):
+
+    # PyORC required to write orc files with a specific
+    # stripe sizes
+    pyorc = pytest.importorskip("pyorc")
+
+    size = 10_000
+    df = pd.DataFrame(
+        {
+            "num": np.random.randint(100, size=size),
+            "text": np.random.choice(["dog", "cat", "bird"], size),
+        }
+    )
+
+    path = str(tmpdir)
+    for i in range(0, size, size // 2):
+        _df = df[i : i + size // 2]
+        with open(os.path.join(path, f"data_{i}.orc"), "wb") as f:
```

Comment:
Nitpick: Not meant as a blocking comment, just an FYI, but using pytest's `tmp_path` fixture (instead of `tmpdir`) can make some of the file handling code here slightly more compact

```diff
diff --git a/dask/dataframe/io/tests/test_orc.py b/dask/dataframe/io/tests/test_orc.py
index a79a527e..cbba806d 100644
--- a/dask/dataframe/io/tests/test_orc.py
+++ b/dask/dataframe/io/tests/test_orc.py
@@ -143,7 +143,7 @@ def test_orc_roundtrip_aggregate_files(tmpdir, split_stripes):
     assert_eq(data, df2, check_index=False)


-def test_orc_roundtrip_aggregate_files_offset(tmpdir):
+def test_orc_roundtrip_aggregate_files_offset(tmp_path):

     # PyORC required to write orc files with a specific
     # stripe sizes
@@ -157,10 +157,9 @@ def test_orc_roundtrip_aggregate_files_offset(tmpdir):
         }
     )

-    path = str(tmpdir)
     for i in range(0, size, size // 2):
         _df = df[i : i + size // 2]
-        with open(os.path.join(path, f"data_{i}.orc"), "wb") as f:
+        with open(tmp_path / f"data_{i}.orc", "wb") as f:
             writer = pyorc.Writer(
                 f,
                 "struct<num:int,text:string>",
@@ -176,6 +175,6 @@ def test_orc_roundtrip_aggregate_files_offset(tmpdir):
     # specifying split_stripes=6 & aggregate_files=True should
     # produce 2 partitions (with the first being larger than
     # the second)
-    df2 = dd.read_orc(path, split_stripes=6, aggregate_files=True)
+    df2 = dd.read_orc(tmp_path, split_stripes=6, aggregate_files=True)
     assert df2.npartitions == 2
     assert len(df2.partitions[0].index) > 5_000
```

Reply:
sounds good!